client/mass/nav.js

```diff
@@ -570,14 +570,32 @@ function setRenderers(self) {
 						const state = self.app.getState()
 						//the terms used in the plots are not always the same for the profile.
 						//Therefore when switching cohorts, it is necessary to delete the plots opened
-						if (state.vocab.dslabel == 'profile')
+						if (state.vocab.dslabel == 'profile') {
```

Comment:
we should not hardcode to a specific dslabel here, maybe do this instead, so need to add a new boolean flag to `sjpp/dataset/sjglobal.profile.ts: cohort.termdb.selectCohort`
```js
if (self.state.termdbConfig.selectCohort.clearFilterOnChange) {
```

Reply:
maybe clearStateOnChange, as it also removes the plots?

Reply:
I changed my suggestion to an array, so a dataset may want to clear the filter but not delete the existing plots. So, profile may want to clear both `['plots', 'filter']`, but another dataset may only want to clear one of them or maybe another part of the state, this approach will be more flexible than a boolean.

Reply:
Solved